scenariolib/event_search.go

```diff
@@ -12,20 +12,19 @@ import (
 
 // SearchEvent a struct representing a search, is defined by a query to execute
 type SearchEvent struct {
-	query string
-	// keyword exists because the query sent to the index may be different than the keyword(s) used to search
+	query       string
+	// keyword exists because the query sent to the index may be diffrent than the keyword(s) used to search
 	keyword     string
 	actionCause string
 	actionType  string
 	logEvent    bool
 	customData  map[string]interface{}
 }
 
-func newSearchEvent(e *JSONEvent, c *Config) (*SearchEvent, error) {
+func newSearchEvent(e *JSONEvent, c *Config, v *Visit) (*SearchEvent, error) {
```

Comment:
Well I am refactoring the bot to also have a `User` object but what I am saying is if the search event needs the language to be initialized, only the language should be added to the call

`func newSearchEvent(e *JSONEvent, c *Config, lang string) (*SearchEvent, error)`

`newSearchEvent(jse, c, v.Language)`